lti_consumer/plugin/views.py

```diff
@@ -501,3 +512,398 @@ def list(self, *args, **kwargs):
+    token = request.POST.get('JWT')
+    jwt = JWT().unpack(token)
+
+    client_id = jwt.payload().get('iss')
+
+    try:
+        lti_config = LtiConfiguration.objects.get(lti_1p3_client_id=client_id)
+    except LtiConfiguration.DoesNotExist as exc:
+        log.error("The iss claim %s is not valid for the LTI 1.3 proctoring start_assessment request.", client_id)
+        raise Http404 from exc
+
+    if not lti_config.lti_1p3_proctoring_enabled:
+        log.info("Proctoring Services for LTIConfiguration with config_id %s are not enabled", lti_config.config_id)
+        return JsonResponse(
+            {'error':
+                f'Proctoring Services for LTIConfiguration with config_id {lti_config.config_id} are not enabled'},
+            status=400,
+        )
+
+    lti_consumer = lti_config.get_lti_consumer()
+
+    # Let's grab the session_data stored in the learner's session. This will need to be compared
+    # against the session_data claim in the JWT token included by the Proctoring Tool in the request. This protects
+    # against CSRF attacks.
+    session_data_key = get_cache_key(app="lti", key="session_data", user_id=request.user.id)
+    cached_session_data = TieredCache.get_cached_response(session_data_key)
+    if cached_session_data.is_found:
+        session_data = cached_session_data.value
+    else:
+        return JsonResponse(
+            {'error': 'The provided session_data claim does not match the anti-CSRF token.'},
+            status=400,
+        )
+
+    # TODO: The resource link should uniquely represent the assessment in the Assessment Platform.
+    #       We SHOULD provide a value for the title attribute.
+    #       It's RECOMMENDED to provide a value for the description attribute.
+    #       The xblock-lti-consumer library does not currently support setting these attributes.
+    resource_link = request.POST.get('resource_link')
+    attempt_number = request.POST.get('attempt_number')
+
+    lti_consumer.set_proctoring_data(
+        attempt_number=attempt_number,
+        session_data=session_data,
+        resource_link=resource_link,
+    )
+
+    # TODO: I hardcoded this to None for right now. We'll need to figure out how to supply user_roles from outside the
+    #       platform.
+    # TODO: LTI Proctoring Services expects that the user role is empty or includes
+    #       "http://purl.imsglobal.org/vocab/lis/v2/membership#Learner", which is a context role.
+    #       lti_consumer.set_user_data uses the regular roles. So, for now, I'm leaving it empty.
+    # Required user claim data
+    lti_consumer.set_user_data(
+        user_id=request.user.id,
+        role=None,
+    )
+
+    # These claims are optional. They are necessary to set in order to properly verify the verified_user claim,
+    # if the Proctoring Tool includes it in the JWT.
+    # TODO: This will need to have additional consideration for PII.
```

Comment:
I don't think there's an issue here because we aren't allowing course teams to configure their own providers. Any available vendor will have a privacy agreement with us.